openwisp_network_topology/api/views.py

```diff
@@ -19,7 +30,40 @@
 Topology = swapper.load_model('topology', 'Topology')
 
 
-class NetworkCollectionView(generics.ListAPIView):
+class HasGetMethodPermission(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        user = request.user
+        app_label = Topology._meta.app_label.lower()
+        model = Topology._meta.object_name.lower()
+        change_perm = f'{app_label}.change_{model}'
+        view_perm = f'{app_label}.view_{model}'
+        if user.is_authenticated:
+            if user.is_superuser or request.method != 'GET':
+                return True
+            if request.method == 'GET' and (
+                user.has_permission(change_perm) or user.has_permission(view_perm)
+            ):
+                return True
+        return False
```

Comment:
I do not understand why you changed this?
Before it was better. Now we're probably going to generate more queries to perform this permission check.